src/javascript/namespace-scanner.ts

```diff
@@ -29,16 +30,133 @@ export class NamespaceScanner implements JavaScriptScanner {
       document: JavaScriptDocument,
       visit: (visitor: Visitor) => Promise<void>) {
     const visitor = new NamespaceVisitor(document);
-    await visit(visitor);
+    const propertyFinder = new NamespacePropertyFinder(document);
+
+    await Promise.all([visit(visitor), visit(propertyFinder)]);
+
+    for (const [namespace, properties] of propertyFinder.properties) {
+      if (visitor.namespaces.has(namespace)) {
+        const ns = visitor.namespaces.get(namespace)!;
+        for (const prop of properties.values()) {
+          ns.properties.set(prop.name, prop);
+        }
+      }
+    }
+
     return {
-      features: Array.from(visitor.namespaces),
+      features: [...visitor.namespaces.values()],
       warnings: visitor.warnings
     };
   }
 }
 
+class NamespacePropertyFinder implements Visitor {
```

Comment:
Why isn't this logic just in NamespaceVisitor? It seems like a lot of extra work to find namespaces twice, the first time just to know that the namespace exists, the second time to find their properties, and then reconcile. Why not apply this logic whenever NamespaceVisitor finds a namespace object?